src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs

```diff
@@ -1111,34 +1113,47 @@ private static IEnumerable<FormatViewDefinition> ViewsOf_System_Management_Autom
                                         $message = ''
                                         $prefix = ''
 
+                                        # Handle case where there is a TargetObject from a Pester `Should` assertion failure and we can show the error at the target rather than the script source
+                                        # Note that in some versions, this is a Dictionary<,> and in others it's a hashtable. So we explicitly cast to a shared interface in the method invocation
+                                        # to force using `IDictionary.Contains`. Hashtable does have it's own `ContainKeys` as well, but if they ever opt to use a custom `IDictionary`, that may not.
+                                        $useTargetObject = $null -ne $err.TargetObject -and
```

Comment:
I cannot find where `$err` is defined before this use. Can you please point me to it?

Reply:
> I cannot find where `$err` is defined before this use. Can you please point me to it?

Yeah it's a little confusing. This is inside the nested function `Get-ConciseViewPositionMessage`. Then look at this 

![image](https://github.com/PowerShell/PowerShell/assets/24977523/bd5f3786-498e-4819-ac98-41b620dd1827)

As labeled:

1. is the ending bracket for the function *definition* of `Get-ConciseViewPositionMessage`
2. is where `$err` is set
3. is where the function is called in the main body of the format view

I kept with the existing style, but at some point it may be worth rewriting some parts of this for maintainability.

Reply:
Thank you!